### ISSUE TITLE
Unknown error in UNet2DModel

### ISSUE THREAD

**Issue by chuangshizhiqiang** [huggingface/diffusers#3630]
### Describe the bug

I try to feed UNet2DModel with (BATCHS, 1, 28, 28) shape sample and (BATCHS,) timestep, but I got 
"RuntimeError: Sizes of tensors must match except in dimension 1. Expected size 8 but got size 7 for tensor number 1
in the list." in "diffusers/models/unet_2d_blocks.py:1726 in forward".

Here is how I define the UNet2DModel:
self.model = UNet2DModel(sample_size=(28,28), in_channels=1, out_channels=1)

Here is how I use UNet2DModel:
xt = xt.to(DEVICE).view(BATCHS, 1, 28, 28)
t = t.to(DEVICE)
t = t.squeeze().float()
noisy_hat = self.model(xt, t)

My diffusers version:
diffusers 0.16.1

Error log details:
![image](https://github.com/huggingface/diffusers/assets/16714437/90adca8c-007f-48d2-9f7b-22fab12bb380)

### Reproduction

import torch
import torchvision
from torchvision import datasets, transforms
from torch.utils.data import DataLoader
import matplotlib.pyplot as plt
import numpy as np
from tqdm import tqdm
from pathlib import Path
import einops
from diffusers import UNet2DModel

if torch.cuda.is_available():
    DEVICE='cuda'
else:
    DEVICE='cpu'

T = 1000 # 时间步长
𝛃 = torch.linspace(0.0001, 0.02, T) # 扩散率
𝛂 = 1 - 𝛃
ᾱ = 𝛂.cumprod(dim=0) # 累乘

def noisy_sample(x0, ᾱ, t=None, num_sample=1):
    n = len(x0)
    if t is None:
        t = torch.randint(0, T, (num_sample, 1,1,1), dtype=torch.long)
    ε = torch.normal(0, 1, x0.shape)[None, ...].repeat(num_sample, 1,1,1)
    ᾱ_t = ᾱ[t].reshape(-1, 1, 1, 1)
    xt = x0 * ᾱ_t.sqrt() + (1-ᾱ_t) * ε
    return xt, t, ε

class my_model(torch.nn.Module):
    def __init__(self):
        super(my_model, self).__init__()
        self.model = UNet2DModel(sample_size=(28,28), in_channels=1, out_channels=1)
        # self.model = UNet2DModel()

    # 实际上的 backward xt->x0
    def forward(self, xt, t):
        t = t.squeeze().float()
        noisy_hat = self.model(xt, t)
        return noisy_hat
model = my_model().to(DEVICE)

# train
criterion = torch.nn.MSELoss()
optimizer = torch.optim.SGD(model.parameters(), lr=0.01)

progress_bar = tqdm(EPOCHS)
for epoch in range(EPOCHS):
    running_loss = 0.0
    for _ in range(EPOCH_STEPS):
        optimizer.zero_grad()
        xt, t, noisy = noisy_sample(image.view(-1, 1*28*28), ᾱ, num_sample=BATCHS)
        xt = xt.to(DEVICE).view(BATCHS, 1, 28, 28)
        t = t.to(DEVICE)
        noisy = noisy.to(DEVICE).view(BATCHS, 1, 28, 28)

        noisy_hat = model(xt, t)
        loss = criterion(noisy, noisy_hat)
        loss.backward()
        optimizer.step()
        running_loss += loss.item()
    progress_bar.set_description(f'[{epoch}] loss: {running_loss / EPOCH_STEPS:.3f}')
    progress_bar.update()
#    print('[%d] loss: %.3f' % (epoch, running_loss / EPOCH_STEPS))
torch.save(model.state_dict(), MODEL_PATH/'model.pth')

### System Info

---------------------------- --------------------
absl-py                      1.4.0
accelerate                   0.18.0
aiofiles                     22.1.0
aiohttp                      3.8.4
aiosignal                    1.3.1
aiosqlite                    0.19.0
anyio                        3.6.2
appdirs                      1.4.4
argon2-cffi                  21.3.0
argon2-cffi-bindings         21.2.0
arrow                        1.2.3
asttokens                    2.2.1
astunparse                   1.6.3
async-lru                    2.0.2
async-timeout                4.0.2
attrs                        22.1.0
axial-positional-embedding   0.2.1
Babel                        2.12.1
backcall                     0.2.0
beautifulsoup4               4.11.1
bleach                       5.0.1
blis                         0.7.9
cachetools                   5.2.0
catalogue                    2.0.8
certifi                      2023.5.7
cffi                         1.15.1
chardet                      4.0.0
charset-normalizer           2.1.1
click                        7.1.2
cloudpickle                  2.2.0
comm                         0.1.3
commonmark                   0.9.1
communities                  2.2.0
contourpy                    1.0.6
cramjam                      2.6.2
cryptography                 38.0.4
cycler                       0.11.0
cymem                        2.0.7
dask                         2022.2.0
datasets                     2.12.0
debugpy                      1.6.7
decorator                    5.1.1
defusedxml                   0.7.1
diffusers                    0.16.1
dill                         0.3.6
docutils                     0.19
easydict                     1.10
einops                       0.4.1
entrypoints                  0.4
execnb                       0.1.5
executing                    1.2.0
fastcore                     1.5.29
fastjsonschema               2.16.2
fastparquet                  2023.2.0
fastprogress                 1.0.3
filelock                     3.10.7
flatbuffers                  23.5.9
fonttools                    4.38.0
fqdn                         1.5.1
frozenlist                   1.3.3
fsspec                       2023.3.0
ftfy                         6.1.1
gast                         0.3.3
gensim                       3.8.3
ghapi                        1.0.3
google-auth                  2.15.0
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
grpcio                       1.54.0
h5py                         2.10.0
huggingface-hub              0.13.3
hupper                       1.11
idna                         3.4
imageio                      2.22.4
importlib-metadata           5.1.0
importlib-resources          5.10.0
ipykernel                    6.23.1
ipython                      7.32.0
ipython-genutils             0.2.0
ipywidgets                   8.0.2
isoduration                  20.11.0
jaraco.classes               3.2.3
jax                          0.4.9
jedi                         0.18.2
jeepney                      0.8.0
Jinja2                       3.1.2
joblib                       1.2.0
json5                        0.9.14
jsonpointer                  2.3
jsonschema                   4.17.3
jupyter                      1.0.0
jupyter_client               8.2.0
jupyter-console              6.4.4
jupyter_core                 5.3.0
jupyter-events               0.6.3
jupyter-lsp                  2.2.0
jupyter_server               2.6.0
jupyter_server_fileid        0.9.0
jupyter_server_terminals     0.4.4
jupyter_server_ydoc          0.8.0
jupyter-ydoc                 0.2.4
jupyterlab                   4.0.1
jupyterlab-pygments          0.2.2
jupyterlab_server            2.22.1
jupyterlab-widgets           3.0.3
keras                        2.12.0
Keras-Preprocessing          1.1.2
keyring                      23.11.0
kiwisolver                   1.4.4
libclang                     14.0.6
llvmlite                     0.39.1
local-attention              1.4.4
locket                       1.0.0
Mako                         1.2.4
Markdown                     3.4.1
MarkupSafe                   2.1.1
matplot                      0.1.9
matplotlib                   3.3.4
matplotlib-inline            0.1.6
mistune                      2.0.4
ml-dtypes                    0.1.0
more-itertools               9.0.0
mpmath                       1.2.1
multidict                    6.0.4
multiprocess                 0.70.14
murmurhash                   1.0.9
mypy-extensions              1.0.0
nbclassic                    0.4.8
nbclient                     0.7.2
nbconvert                    7.4.0
nbdev                        2.3.12
nbformat                     5.8.0
neo4j                        5.6.0
neo4j-driver                 4.4.2
nest-asyncio                 1.5.6
networkx                     2.4
nltk                         3.5
node2vec                     0.3.3
notebook                     6.5.4
notebook_shim                0.2.2
numba                        0.56.4
numpy                        1.23.5
nvidia-cublas-cu11           11.10.3.66
nvidia-cuda-nvrtc-cu11       11.7.99
nvidia-cuda-runtime-cu11     11.7.99
nvidia-cudnn-cu11            8.5.0.96
oauthlib                     3.2.2
opencv-python                4.6.0.66
opt-einsum                   3.3.0
overrides                    7.3.1
packaging                    21.3
pandas                       1.5.3
pandocfilters                1.5.0
parso                        0.8.3
partd                        1.3.0
PasteDeploy                  3.0.1
pathy                        0.10.1
patool                       1.12
patsy                        0.5.3
pexpect                      4.8.0
pickleshare                  0.7.5
Pillow                       9.3.0
pip                          22.2.2
pkginfo                      1.9.2
pkgutil_resolve_name         1.3.10
plaster                      1.1.2
plaster-pastedeploy          1.0.1
platformdirs                 2.5.4
preshed                      3.0.8
product-key-memory           0.1.10
progressbar2                 4.2.0
prometheus-client            0.15.0
prompt-toolkit               3.0.33
protobuf                     4.23.0
psutil                       5.9.4
ptyprocess                   0.7.0
pure-eval                    0.2.2
pyarrow                      11.0.0
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycparser                    2.21
pycuda                       2022.2
pydantic                     1.7.4
pydot                        1.4.2
Pygments                     2.13.0
pyloco                       0.0.139
pyparsing                    3.0.9
pyramid                      2.0.1
pyre-extensions              0.0.30
pyrsistent                   0.19.2
python-dateutil              2.8.2
python-json-logger           2.0.7
python-utils                 3.4.5
pytools                      2022.1.13
pytz                         2022.6
PyWavelets                   1.4.1
PyYAML                       6.0
pyzmq                        24.0.1
qtconsole                    5.4.0
QtPy                         2.3.0
rapids                       0.0.1
readme-renderer              37.3
reformer-pytorch             1.4.4
regex                        2022.10.31
requests                     2.28.1
requests-oauthlib            1.3.1
requests-toolbelt            0.10.1
responses                    0.18.0
rfc3339-validator            0.1.4
rfc3986                      2.0.0
rfc3986-validator            0.1.1
rich                         12.6.0
rsa                          4.9
sacremoses                   0.0.53
scikit-image                 0.19.3
scikit-learn                 0.24.0
scipy                        1.8.0
seaborn                      0.12.2
SecretStorage                3.3.3
Send2Trash                   1.8.0
sentry-sdk                   1.16.0
setuptools                   65.5.0
SimpleWebSocketServer        0.1.2
six                          1.15.0
sktime                       0.4.1
smart-open                   6.3.0
sniffio                      1.3.0
soupsieve                    2.3.2.post1
spacy                        3.0.3
spacy-legacy                 3.0.12
srsly                        2.4.6
stack-data                   0.6.2
statsmodels                  0.13.5
stellargraph                 1.2.1
sympy                        1.11.1
tensorboard                  2.12.3
tensorboard-data-server      0.7.0
tensorboard-plugin-wit       1.8.1
tensorflow                   2.12.0
tensorflow-addons            0.19.0
tensorflow-estimator         2.12.0
tensorflow-io-gcs-filesystem 0.28.0
tensorlayerx                 0.5.7
termcolor                    1.1.0
terminado                    0.17.1
Theano                       1.0.5
thinc                        8.0.17
thop                         0.1.1.post2209072238
threadpoolctl                3.1.0
tifffile                     2022.10.10
tinycss2                     1.2.1
tokenizers                   0.13.2
tomli                        2.0.1
toolz                        0.12.0
torch                        1.13.1
torch-geometric              2.3.0
torch-scatter                2.1.1
torcheval                    0.0.6
torchtnt                     0.1.0
torchvision                  0.14.1
tornado                      6.2
tqdm                         4.64.0
traitlets                    5.6.0
transformers                 4.28.1
translationstring            1.4
twine                        4.0.2
typeguard                    2.13.3
typer                        0.3.2
typing_extensions            4.6.2
typing-inspect               0.8.0
ultralytics                  8.0.50
uri-template                 1.2.0
urllib3                      1.26.13
ushlex                       0.99.1
venusian                     3.0.0
wasabi                       0.10.1
watchdog                     3.0.0
wcwidth                      0.2.5
webcolors                    1.13
webencodings                 0.5.1
WebOb                        1.8.7
websocket-client             1.4.2
Werkzeug                     2.2.2
wheel                        0.37.1
widgetsnbextension           4.0.3
wrapt                        1.12.1
xeus-python                  0.15.8
xeus-python-shell            0.5.0
xxhash                       3.2.0
y-py                         0.5.9
yarl                         1.9.2
ypy-websocket                0.8.2
zipp                         3.11.0
zope.deprecation             5.0
zope.interface               6.0

**Comment by sayakpaul** [comment]
Is it possible to provide a Colab Notebook that reproduces the error? The snippet you provided is a little hard to parse.

**Comment by chuangshizhiqiang** [comment]
> Is it possible to provide a Colab Notebook that reproduces the error? The snippet you provided is a little hard to parse.

Can you open this Colab Notebook?

https://colab.research.google.com/drive/1PyZ4JdNrTeGd778gOOTY4N6tjjUFGle6?usp=sharing

**Comment by sayakpaul** [comment]
There seems to be an issue in how you initialize the model and call it afterward. 

You can check out the test suite for `UNet2DModel` to get an example of how that should be done. 

But here's an example: https://colab.research.google.com/gist/sayakpaul/04dfc2b9af5cbfe37b80fdd2a70503c5/scratchpad.ipynb

**Comment by chuangshizhiqiang** [comment]
> There seems to be an issue in how you initialize the model and call it afterward.
> 
> You can check out the test suite for `UNet2DModel` to get an example of how that should be done.
> 
> But here's an example: https://colab.research.google.com/gist/sayakpaul/04dfc2b9af5cbfe37b80fdd2a70503c5/scratchpad.ipynb

Thanks for the example, my bad. I miss the sample Dimensions must be a multiple of 2 ** (len(block_out_channels) - 1).
